internal/weldr/api.go

```diff
@@ -43,15 +44,16 @@ type API struct {
 	store   *store.Store
 	workers *worker.Server
 
-	rpmmd  rpmmd.RPMMD
-	arch   distro.Arch
-	distro distro.Distro
-	repos  []rpmmd.RepoConfig
+	rpmmd rpmmd.RPMMD
 
 	logger *log.Logger
 	router *httprouter.Router
 
 	compatOutputDir string
+
+	hostDistroName string                                   // Name of the host distro
+	distros        *distroregistry.Registry                 // Supported distros
+	distroRepos    map[string]map[string][]rpmmd.RepoConfig // Repos for each distro
```

Comment:
Out of curiosity, what's the motivation for keeping repositories for all architectures? Is it a potential future extension to allow specifying also architecture in the Blueprint and allowing building images on remote workers running on a different architecture?

Reply:
Both that, and because the architecture code may differ between the distros. Each one has it's own code so in theory the could be different between distro/release.

Reply:
Right, but to your last point, this is not actually anyhow reflected in the code. The architecture code for all distros is determined by calling `common.CurrentArch()` which effectively returns always the same arch code not based on what the distro uses.

Reply:
The host architecture name is from `common.CurrentArch()`, but if you look at the code in `internal/distro/*/distro.go there are different implementations for each distro+arch. `LoadRepositories()` already returns a map of the per-arch repos so we may as well keep them all, even if we don't use them yet.